packages/modals/src/elements/DrawerModal/DrawerModal.tsx

```diff
@@ -49,12 +58,32 @@ const DrawerModalComponent = forwardRef<HTMLDivElement, IDrawerModalProps>(
       useModal({
         idPrefix: id,
         modalRef,
-        focusOnMount,
-        restoreFocus,
+        focusOnMount: false /* [1:b] */,
+        restoreFocus: false /* [1:b] */,
         environment,
         onClose
       });
 
+    /* [1:c] */
+    useEffect(() => {
+      if (environment) {
+        if (modalRef.current && isOpen) {
+          if (restoreFocus === undefined ? true : restoreFocus) {
+            triggerRef.current = activeElement(environment) as HTMLElement;
+          }
+
+          if (focusOnMount === undefined ? true : focusOnMount) {
+            modalRef.current.focus();
+          }
+        }
+
+        if (triggerRef.current && !isOpen) {
```

Comment:
What happens if `modalRef`, `triggerRef`, `restoreFocus`, or `focusOnMount` change but neither `triggerRef` nor `isOpen` does and this condition still passes?

Reply:
I'm also wondering if this should include a `restoreFocus` check.

Reply:
> What happens if `modalRef`, `triggerRef`, `restoreFocus`, or `focusOnMount` change but neither `triggerRef` nor `isOpen` does and this condition still passes?

That is a good question, let me confirm with additional test cases.

> I'm also wondering if this should include a `restoreFocus` check.

I don't think we necessarily need it. In this case, `triggerRef.current` would only be assigned if `restoreFocus` is `true`, which is why it's not necessary to check since the triggerRef indicates that focus was moved away from it when the modal was opened.

Reply:
I see - so if another ref changes for some reason, it's pretty safe to assume it's due to modal closing and not something else?

Reply:
In this case, the `isOpen` prop is what mainly drives this effect and as that is toggled, it would used to point the focus where it should be. With React refs are attached before effects are ran so we can safely assume it is present when `CSSTransition` mounts the `StyledDrawer` to the DOM. I realized after we don't need to add the refs to the dependency array so I removed them.